src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -0,0 +1,513 @@
+        {
+            // check if same topics/partitions have already been requested by someone else,
+            // which is an indication that something is wrong, but it might be alright when assignments change.
+            foreach (var (actorRef, request) in _requests.ToTuples())
+            {
+                if (!actorRef.Equals(fromStage) && request.Topics.Any(topics.Contains))
+                {
+                    _log.Warning($"{updateType} from topic/partition {string.Join(", ", topics)} " +
+                                 $"already requested by other stage {string.Join(", ", request.Topics)}");
+                    actorRef.Tell(new KafkaConsumerActorMetadata.Internal.Messages<K, V>(request.RequestId, ImmutableList<ConsumeResult<K, V>>.Empty));
+                    _requests = _requests.Remove(actorRef);
+                }
+            }
+        }
+
+        private void ReceivePoll(Internal.Poll<K, V> poll)
+        {
+            if (poll.Target == this)
+            {
+                var refreshOffsets = _commitRefreshing.RefreshOffsets;
+                if (refreshOffsets.Any())
+                {
+                    _log.Debug($"Refreshing comitted offsets: {refreshOffsets.JoinToString(", ")}");
+                    Commit(refreshOffsets, msg => Context.System.DeadLetters.Tell(msg));
+                }
+               
+                Poll();
+               
+                if (poll.Periodic)
+                    SchedulePoolTask();
+                else
+                    _delayedPoolInFlight = false;
+            }
+            else
+            {
+                // Message was enqueued before a restart - can be ignored
+                _log.Debug("Ignoring Poll message with stale target ref");
+            }
+        }
+
+        private void Poll()
+        {
+            var currentAssignment = _consumer.Assignment;
+            var initialRebalanceInProcess = _rebalanceInProgress;
+
+            try
+            {
+                if (_requests.IsEmpty())
+                {
+                    // no outstanding requests so we don't expect any messages back, but we should anyway
+                    // drive the KafkaConsumer by polling to handle partition events etc.
+                    // TODO: Make consumer pause-resume calls work
+                    // _consumer.Pause(currentAssignment);
+                    var message = _consumer.Consume(TimeSpan.FromMilliseconds(1));
+                    if (message != null)
+                        throw new IllegalActorStateException("Got unexpected Kafka message");
+                }
+                else
+                {
+                    // resume partitions to fetch
+                    IImmutableSet<TopicPartition> partitionsToFetch = _requests.Values.SelectMany(v => v.Topics).ToImmutableHashSet();
+                    /*
+                    // TODO: Make consumer pause-resume calls work
```

Comment:
@Aaronontheweb This is a most tricky thing left in this actor.
For some reason, if I will uncomment